src/storage-types/src/sources.rs

```diff
@@ -133,7 +132,9 @@ impl<S: Debug + Eq + PartialEq> IngestionDescription<S> {
 
         let compatibility_checks = [
```

Comment:
nit: I think the checks with desc like `(remap_shard == &other.remap_shard, "remap_shard"),` above was nice. Is that possible to do it here?

Reply:
Oh yes, I have all of those in another PR and didn't port them over here, but I should.